src/logging.js

```diff
@@ -133,14 +133,42 @@ export async function makeRequestMiddleware(
 ): Promise<Middleware> {
     // This is the logger that captures requests handled by our express server.
     return args.dev
-        ? (req: $Request, res: $Response, next: NextFunction) => {
-              // $FlowIgnore: We make this up for google
-              req.log = logger;
-              next();
-          }
-        : await lw.express.makeMiddleware(logger);
+        ? /**
+           * If we're in dev, we're going to use the expressWinston logger
+           */
+          Promise.resolve(
+              expressWinston.logger({
+                  /**
+                   * Specify the level that this logger logs at.
+                   * (use a function to dynamically change level based on req and res)
+                   *     `function(req, res) { return String; }`
+                   */
+                  level: "info",
+
+                  /**
+                   * Use the logger we already set up.
+                   */
+                  winstonInstance: logger,
+                  expressFormat: true,
+                  colorize: true,
+                  meta: false,
+              }),
+          )
+        : /**
+           * Otherwise, we're using the Google middleware
+           */
+          await lw.express.makeMiddleware(logger);
 }
 
-const logger: Logger = initLogging(args.logLevel, args.dev);
-
-export default logger;
+export const rootLogger: Logger = initLogging(args.logLevel, args.dev);
+
+let scopedLogger: ?Logger = null;
+export const getScopedLogger = (): Logger => {
+    return scopedLogger != null ? scopedLogger : rootLogger;
+};
+export const useScopedLogger = (logger: Logger) => {
+    scopedLogger = logger;
+};
+export const revertScopedLogger = () => {
+    scopedLogger = null;
+};
```

Comment:
Since this is set on the module level, I'm assuming we don't have any async code in the request handler where we might process multiple requests at once asynchronously? If that's the case, it'd be good to make sure that's documented somewhere as a requirement. Otherwise future changes might make this a bit of a mess if we've got logs being added to the wrong requests.

Reply:
That's a good callout. I was pondering on that too. I'm not sure how NodeJS/express handles the concept of concurrency given that JS is traditionally expected to be single-threaded with only very specific actions being farmed off to threading behind the scenes (like fetches).

I'll poke at this a bit.

Reply:
This may actually relate to an issue I am investigating.

Reply:
OK, I've reworked this to pass around the logging as I think that it's too fragile the way I had it.